abjadext/nauert/qevents.py

```diff
@@ -121,73 +87,21 @@ class PitchedQEvent(QEvent):
 
     ### CLASS VARIABLES ###
 
-    __slots__ = ("_attachments", "_index", "_offset", "_pitches")
-
-    _publish_storage_format = True
+    pitches: typing.Optional[
+        typing.Iterable[typing.Union[numbers.Number, abjad.typings.Number]]
+    ] = None
+    attachments: typing.Optional[typing.Iterable] = None
 
     ### INITIALIZER ###
 
-    def __init__(
-        self,
-        offset: typing.Union[numbers.Real, abjad.typings.Number, abjad.IntegerPair] = 0,
-        pitches: typing.Optional[
```

Comment:
Yes, we can usually list the fields in any order that we want, but with derived classes, it will be a bit tricky.

```python
import dataclasses


@dataclasses.dataclass
class Fruit:
    a: int
    c: int

@dataclasses.dataclass
class Lime(Fruit):
    b: int

fruit = Fruit(1, 3) # Fruit(a=1, c=3)
lime = Lime(1, 2, 3) # Lime(a=1, c=2, b=3)
```

Adding the inherited fields in the derived class does not really help either:
```python
import dataclasses


@dataclasses.dataclass
class Fruit:
    a: int
    c: int

@dataclasses.dataclass
class Lime(Fruit):
    a: int
    b: int
    c: int

fruit = Fruit(1, 3) # Fruit(a=1, c=3)
lime = Lime(1, 2, 3) # Lime(a=1, c=2, b=3)
```